[topic-gpio] Convert bbc:microbit samples to the new GPIO API

carlescufi opened the pull request:
The `line_follower_robot` sample is untested on hardware, since I do not have it.

carlescufi committed:
samples: micro:bit: Convert sound sample to new GPIO API

Convert to the new GPIO API, tested on bbc:microbit.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

carlescufi committed:
samples: micro:bit: Convert pong sample to new GPIO API

Convert to the new GPIO API, tested on bbc:microbit.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

carlescufi committed:
samples: micro:bit: Convert line_follower_robot sample to new GPIO API

Convert to the new GPIO API, untested on hardware.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

carlescufi commented:
@jhedberg do you have the means to test the line follower robot? I tested the other 2.

mnkp reviewed (commented):
In `samples/boards/bbc_microbit/line_follower_robot/src/main.c` the `line_detection` function contains calls to `gpio_pin_read`. They should be changed to `gpio_pin_get_raw`. The `_raw` version seems more appropriate as `gpio_pin_configure` function doesn't include DTS flags / doesn't take into account pin active level.

carlescufi commented:
> In `samples/boards/bbc_microbit/line_follower_robot/src/main.c` the `line_detection` function contains calls to `gpio_pin_read`. They should be changed to `gpio_pin_get_raw`. The `_raw` version seems more appropriate as `gpio_pin_configure` function doesn't include DTS flags / doesn't take into account pin active level.

Agreed, done.

anangl commented on samples/boards/bbc_microbit/pong/src/main.c:
```diff
@@ -483,23 +483,27 @@ void pong_remote_lost(void)
 	gpio_pin_configure(gpio, DT_ALIAS_SW0_GPIOS_PIN,
-			   (GPIO_DIR_IN | GPIO_INT | GPIO_INT_EDGE |
-			    GPIO_INT_ACTIVE_LOW));
+			   DT_ALIAS_SW0_GPIOS_FLAGS | GPIO_INPUT);
```
Flags in `bbc_microbit.dts` should be also updated for this to work: `GPIO_INT_ACTIVE_LOW` -> `GPIO_ACTIVE_LOW`.

carlescufi replied:
Oh yes, well spotted! This seemed to work because the button press was being detected on release of the button, so it fooled me.

carlescufi committed:
boards: bbc_microbit: Convert GPIO flags to new API

Use the GPIO_ACTIVE_LOW to signal that the button pins are active low,
instead of the deprecated GPIO_INT_ACTIVE_LOW.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>
